### PR TITLE
Created Teaser Component

### PR DESCRIPTION
**Pull request by stenizwahyudi**
Description
As a front end developer, I want to create a Teaser component; the entire component should be linked and it should include the following fields:
1. URL
2. Title
3. Small Text

The teaser comes in three styles:
1. With a Dark Teal Line
2. With a Teal Line
3. With a Gray Line and Gray Line (when the link is disabled)

Each Teaser has a light gray bottom border and an arrow icon which can be found on Google Material Design. Icon files should be named according to the following convention: icon-{descriptive-name}.svg

Sketch: https://www.sketch.com/s/3e6b9c91-118c-471e-98ad-8676d41d1435/a/zxyGnLW

### CODE REVIEW

File: src/stories/components/Teaser/styles.scss [hunk 1]
```diff
@@ -0,0 +1,56 @@
+@import '../../../scss/variables.scss';
+
+// @file
+// Teaser
+
+.teaser {
+    width: 100%;
+    border: solid lightgrey;
```
Comment (stenizwahyudi) [inline src/stories/components/Teaser/styles.scss:8]: The color from the global variable doesn't match the sketch, so I used lightgrey instead.

File: src/stories/components/Teaser/styles.scss [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }
+}
+
+.teaser--disabled {
+    color: grey;
+    .arrow__logo {
+        visibility: hidden;
+    }
+    .line {
+        border: solid grey;
```
Comment (stenizwahyudi) [inline src/stories/components/Teaser/styles.scss:54]: The color from the global variable doesn't match the sketch, so I used grey instead.

File: src/stories/components/Teaser/Teaser.stories.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    url: '#',
+    title: 'Title with a Teal Line',
+    text: 'Teal line',
+};
+
+// Disabled teaser
+export const Disabled = Template.bind({});
+Disabled.args = {
+    modifierClasses: 'teaser--disabled',
+    url: '#',
```
Comment (cmutuc25) [inline src/stories/components/Teaser/Teaser.stories.js:36]: The disabled teaser should probably not link somewhere. Maybe disable the link and keep the url to pass in?

**Review by cmutuc25 — COMMENTED** [review]
Left a comment about the disabled type

File: src/stories/components/Teaser/Teaser.stories.js (hunk 3 above)
Comment (stenizwahyudi) [reply src/stories/components/Teaser/Teaser.stories.js:36]: That is a very good observation.

Comment (stenizwahyudi) [reply src/stories/components/Teaser/Teaser.stories.js:36]: I disabled the link redirection in the styles.scss (see line 49)

File: src/stories/components/Teaser/styles.scss (hunk 2 above)
Comment (alecherryy) [reply src/stories/components/Teaser/styles.scss:55]: I'll double check this. The repo might be out of date or the Sketch file is wrong.

**Review by alecherryy — COMMENTED** [review]
@stenizwahyudi all looks good, except for the `className`'s. Don't forget to use the BEM convention; this way you don't have to nest elements to bind the styles. 

I'm gonna update the component to include a fun hover animation.

File: src/stories/components/Teaser/Teaser.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    return (
+        <a
+            href={url}
+            className={['teaser', `${modifierClasses}`].join(' ').trim()}
+        >
+            <img src={Logo}
+                alt={`${text} arrow logo`}
+                className="arrow__logo"
+            />
+            <div className='line'></div>
```
Comment (alecherryy) [inline src/stories/components/Teaser/Teaser.js:31]: Rather than adding extra markup, you can use `pseudo-elements` to add decorative elements.

Comment (stenizwahyudi) [reply src/stories/components/Teaser/Teaser.js]: Aha, I just knew about this. Done, thank you!

**Review by alecherryy — APPROVED** [review]

**Pull request merged**